editor/reducer.js

```diff
@@ -454,6 +422,36 @@ export function preferences( state = STORE_DEFAULTS.preferences, action ) {
 				...state,
 				mode: action.mode,
 			};
+		case 'INSERT_BLOCKS':
+			// record the block usage and put the block in the recently used blocks
+			let blockUsage = state.blockUsage;
+			let recentlyUsedBlocks = [ ...state.recentlyUsedBlocks ];
+			action.blocks.forEach( ( block ) => {
+				const uses = ( blockUsage[ block.name ] || 0 ) + 1;
+				blockUsage = omit( blockUsage, block.name );
+				blockUsage[ block.name ] = uses;
+				recentlyUsedBlocks = [ block.name, ...without( recentlyUsedBlocks, block.name ) ].slice( 0, maxRecent );
+			} );
+			return {
+				...state,
+				blockUsage,
+				recentlyUsedBlocks,
+			};
+		case 'SETUP_EDITOR':
+			const filterInvalidBlocksFromList = list => list.filter( name => getBlockType( name ) !== undefined );
```

Comment:
Minor: This could be factorized `const isBlockDefined = name => getBlockType( name ) !== undefined` and reused in both `filterInvalidBlocksFromList` and `filterInvalidBlocksFromObject`
I don't know if it makes it clearer though

Reply:
I like it, certainly makes it explicit what is being checked for, instead of duplicating the logic of how we check. Fixed!